### ISSUE TITLE
bug: two phase stream agg panicked at `Option::unwrap()`

### ISSUE THREAD

**Issue by kwannoel** [risingwavelabs/risingwave#7928]
### Describe the bug

```
2022-08-08T16:38:57.971800Z  INFO node{id=14 name="client"}:task{id=1339}: risingwave_sqlsmith::runner: Executing: SET RW_FORCE_TWO_PHASE_AGG TO TRUE
2022-08-08T16:38:58.030585Z  INFO node{id=14 name="client"}:task{id=1339}: risingwave_sqlsmith::runner: Executing: CREATE MATERIALIZED VIEW stream_query AS SELECT (37) AS col_0, (INTERVAL '-60') AS col_1, false AS col_2, CAST(NULL AS STRUCT<a NUMERIC>) AS col_3 FROM (SELECT false AS col_0, (SMALLINT '255') AS col_1, t_0.l_commitdate AS col_2 FROM lineitem AS t_0 GROUP BY t_0.l_returnflag, t_0.l_suppkey, t_0.l_tax, t_0.l_shipinstruct, t_0.l_orderkey, t_0.l_commitdate, t_0.l_quantity HAVING true) AS sq_1 WHERE false GROUP BY sq_1.col_2
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/frontend/src/optimizer/property/distribution.rs:254:86
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:64:14
   2: core::panicking::panic
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/panicking.rs:114:5
   3: core::option::Option<T>::unwrap
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/option.rs:823:21
   4: risingwave_frontend::optimizer::property::distribution::DistributionDisplay::fmt
             at ./src/frontend/src/optimizer/property/distribution.rs:254:86
   5: core::fmt::builders::DebugStruct::field::{{closure}}
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/builders.rs:141:17
   6: core::result::Result<T,E>::and_then
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/result.rs:1371:22
   7: core::fmt::builders::DebugStruct::field
             at /rustc/3984bc5833db8bfb0acc522c9775383e4171f3de/library/core/src/fmt/builders.rs:124:23
   8: <risingwave_frontend::optimizer::plan_node::stream_exchange::StreamExchange as core::fmt::Display>::fmt
             at ./src/frontend/src/optimizer/plan_node/stream_exchange.rs:53:9
   9: core::fmt::write
```

### To Reproduce

```
set rw_force_two_phase_agg to true;                       
CREATE TABLE lineitem (
    l_orderkey BIGINT,
    l_partkey INTEGER,
    l_suppkey INTEGER,
    l_linenumber INTEGER,
    l_quantity NUMERIC,
    l_extendedprice NUMERIC,
    l_discount NUMERIC,
    l_tax NUMERIC,
    l_returnflag VARCHAR,
    l_linestatus VARCHAR,
    l_shipdate DATE,
    l_commitdate DATE,
    l_receiptdate DATE,
    l_shipinstruct VARCHAR,
    l_shipmode VARCHAR,
    l_comment VARCHAR,
    PRIMARY KEY (l_orderkey, l_linenumber)
);


CREATE MATERIALIZED VIEW stream_query AS SELECT (37) AS col_0, (INTERVAL '-60') AS col_1, false AS col_2, CAST(NULL AS STRUCT<a NUMERIC>) AS col_3 FROM (SELECT false AS col_0, (SMALLINT '255') AS col_1, t_0.l_commitdate AS col_2 FROM lineitem AS t_0 GROUP BY t_0.l_returnflag, t_0.l_suppkey, t_0.l_tax, t_0.l_shipinstruct, t_0.l_orderkey, t_0.l_commitdate, t_0.l_quantity HAVING true) AS sq_1 WHERE false GROUP BY sq_1.col_2;
```

### Additional context

Related to `RW_FORCE_TWO_PHASE_AGG`.

**Comment by jon-chuang** [comment]
Is this a global simple agg? If so, I think it may be fixed by https://github.com/risingwavelabs/risingwave/pull/7932?

**Comment by kwannoel** [comment]
> Is this a global simple agg? If so, I think it may be fixed by #7932?

Error occurs in frontend, seems like caused by wrong group key in exchange. Still fixing it
